dace/codegen/targets/xilinx.py

```diff
@@ -193,19 +200,75 @@ def define_stream(dtype, vector_length, buffer_size, var_name, array_size,
             kernel_stream.write("dace::SetNames({}, \"{}\", {});".format(
                 var_name, var_name, cpp.sym2cpp(array_size)))
 
-    @staticmethod
-    def define_local_array(dtype, vector_length, var_name, array_size, storage,
-                           shape, function_stream, kernel_stream, sdfg,
-                           state_id, node):
+    def define_local_array(self, var_name, desc, array_size, veclen,
+                           function_stream, kernel_stream, sdfg, state_id,
+                           node):
         kernel_stream.write("dace::vec<{}, {}> {}[{}];\n".format(
-            dtype.ctype, vector_length, var_name, cpp.sym2cpp(array_size)))
-        if storage == dace.dtypes.StorageType.FPGA_Registers:
-            kernel_stream.write("#pragma HLS ARRAY_PARTITION variable={} "
-                                "complete\n".format(var_name))
-        elif len(shape) > 1:
-            kernel_stream.write("#pragma HLS ARRAY_PARTITION variable={} "
-                                "block factor={}\n".format(
-                                    var_name, shape[-2]))
+            desc.dtype.ctype, veclen, var_name, cpp.sym2cpp(array_size)))
+        if desc.storage == dace.dtypes.StorageType.FPGA_Registers:
+            kernel_stream.write(
+                "#pragma HLS ARRAY_PARTITION variable={} "
+                "complete\n".format(var_name), node, state_id, sdfg)
+        elif len(desc.shape) > 1:
+            kernel_stream.write(
+                "#pragma HLS ARRAY_PARTITION variable={} "
+                "block factor={}\n".format(var_name, desc.shape[-2]), node,
+                state_id, sdfg)
+        self._dispatcher.defined_vars.add(var_name, DefinedType.Pointer)
+
+    def _find_shift_register_accesses(self, sdfg, dfg, node, shift_width):
+        name = node.data
+        accesses = set()
+        to_search = list(dfg.out_edges(node))
+        unrolled_indices = {}
+        constants = sdfg.constants
+        for k, v in constants.items():
+            unrolled_indices[k] = [v]
+        while len(to_search) > 0:
+            e = to_search.pop()
+            if isinstance(
+                    e.dst,
+                (dace.graph.nodes.Tasklet, dace.graph.nodes.AccessNode)):
```

Comment:
might be easier to ask if it's not an instance of an entry/exit node

Reply:
This is not actually used yet. I was gonna save it for later, but I can also remove it?

Reply:
Removed it